atomicapp/nulecule/config.py

```diff
@@ -0,0 +1,123 @@
+import copy
+import logging
+
+from atomicapp.constants import (GLOBAL_CONF,
+                                 LOGGER_COCKPIT,
+                                 DEFAULT_PROVIDER,
+                                 DEFAULT_ANSWERS)
+from collections import defaultdict
+
+cockpit_logger = logging.getLogger(LOGGER_COCKPIT)
+
+
+class Config(object):
+    """
+    This class allows to store config data in different scopes along with
+    source info for the data. When fetching the value for a key in a scope,
+    the source info and the PRIORITY order of sources is taken into account.
+    """
+
```

Comment:
Can we get a summary of each `source` below in a comment. This would help future readers of the code.